package.json

```diff
@@ -192,6 +190,10 @@
     "watchify": "3.11.0",
     "yamljs": "0.3.0"
   },
+  "optionalDependencies": {
+    "@percy/puppeteer" : "0.4.0",
+    "puppeteer" : "1.9.0"
+  },
```

Comment:
I think these will still get installed when `yarn` is run without any args. (Our [docs](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server) instruct developers to run `yarn` (without any args) during first time install.)

I wonder if there's a way to make it so that these dependencies aren't expressed in `package.json`, but still get installed when `gulp visual-diff` is run.

If this turns out to be difficult / infeasible, I think we're still good, since the automated `gulp update-packages` will still run faster because you're not installing optional dependencies.

Reply:
One way to move the packages out of `package.json` and unblock the `yarn` critical path might be to do a global install of `puppeteer` and `percy` instead of an in-repo install (`yarn global add puppeteer percy`). However, this will result in us losing the ability to get automatic updates, so I'm not a huge fan. Any other ideas @danielrozenberg?

Reply:
It's possible to call `yarn install <…>` instead of `npx yarn --force`, but removing them from `package.json` means Renovate bot won't be able to update them for us. I think it's better to keep them as optionalDependencies here

Reply:
Is it ok to change `npx yarn --force` to `yarn install --force`? It's redundant because `gulp update-packages` runs right before it, but without `--force` Yarn doesn't pick up that you're missing the optional dependencies.

Reply:
We use `npx yarn` because it works in the absence of a global installation of `yarn` (useful for external developers who don't use `yarn` for their other projects).